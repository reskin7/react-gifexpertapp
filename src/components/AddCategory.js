import React, { useState } from 'react';
import PropTypes from 'prop-types';

export const AddCategory = ({ setCategories, setQuantity }) => {
  const [inputValue, setInputValue] = useState('');

  const handleInputChange = (event) => {
    setInputValue(event.target.value);
  };

  const handleSubmit = (event) => {
    event.preventDefault();

    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue('');
    }
  };

  const handleQuantity = (event) => {
    setQuantity(event.target.value);
  };

  return (
    <form onSubmit={handleSubmit}>
      <input type="text" value={inputValue} onChange={handleInputChange} />
      <br />
      <br />
      <label htmlFor="quantity">
        Selecciona la cantidad de imagenes que quieres traer {'\n \n'}
      </label>
      <select name="selector" id="quantity" onChange={handleQuantity}>
        <option value={1}>1</option>
        <option value={3}>3</option>
        <option value={5}>5</option>
        <option value={10}>10</option>
      </select>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
