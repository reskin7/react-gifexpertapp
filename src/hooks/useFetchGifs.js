import { useEffect, useState } from 'react';
import { getGifs } from '../helpers/getGifs';
export const useFetchGifs = (category, quantity) => {
  const [state, setState] = useState({
    data: [],
    loading: true,
  });

  useEffect(() => {
    getGifs(category, quantity).then((imgs) => {
      setState({
        data: imgs,
        loading: false,
      });
    });
  }, [category]); // eslint-disable-line react-hooks/exhaustive-deps

  return state;
};
