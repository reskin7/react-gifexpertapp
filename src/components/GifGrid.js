import React from 'react';
import { useFetchGifs } from '../hooks/useFetchGifs';
// import { getGifs } from '../helpers/getGifs';
import { GifGridItem } from './GifGridItem';

export const GifGrid = ({ category, quantity }) => {
  // const [images, setImages] = useState([]);
  const { data: images, loading } = useFetchGifs(category, quantity); // "data: images" es para renombrar...."data" lo voy a poder usar como si fuera una variable de nombre "images"

  // useEffect(() => {
  //   getGifs(category).then(setImages);
  // }, [category]);

  return (
    <>
      <h3 className="animate__animated animate__fadeIn">{category}</h3>

      {loading && <p className="animate__animated animate__flash">Loading</p>}
      <div className="card-grid">
        {images.map((img) => {
          return <GifGridItem key={img.id} {...img} />;
        })}
      </div>
    </>
  );
};
