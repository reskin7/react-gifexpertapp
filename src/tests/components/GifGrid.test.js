import { shallow } from 'enzyme';
import { GifGrid } from '../../components/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
import '@testing-library/jest-dom';

jest.mock('../../hooks/useFetchGifs');

describe('Test en GifGrid', () => {
  const category = 'Microsoft';

  test('should match the snapshot ', () => {
    useFetchGifs.mockReturnValue({
      data: [],
      loading: true,
    });
    const wrapper = shallow(<GifGrid category={category} quantity={2} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('should show items when images are loaded with useFetchGifs', () => {
    const gifs = [
      {
        id: 'ABC',
        url: 'http://www.google.com/HOLA.jpg',
        title: 'TITULO 1',
      },
      {
        id: 'DFG',
        url: 'http://www.google.com/CHAU.jpg',
        title: 'TITULO 2',
      },
    ];
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false,
    });
    const wrapper = shallow(<GifGrid category={category} quantity={2} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length);
  });
});
