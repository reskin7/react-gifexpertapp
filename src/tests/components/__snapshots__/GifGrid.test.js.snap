// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test en GifGrid should match the snapshot  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Microsoft
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Test en GifGrid should show items when images are loaded with useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    Microsoft
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="TITULO 1"
      url="http://www.google.com/HOLA.jpg"
    />
    <GifGridItem
      id="DFG"
      key="DFG"
      title="TITULO 2"
      url="http://www.google.com/CHAU.jpg"
    />
  </div>
</Fragment>
`;
